### PR TITLE
fix: Add in missing transforms from options in find calls

### PR DESCRIPTION
**Pull request by fumbleforce**
Should fix issue #18 

It also fixes a selector issue in update, when the selector is a string.

### CODE REVIEW

File: lib/redis/raceDetectionManager.js [hunk 1]
```diff
@@ -43,8 +43,10 @@ export const onMutation = (collectionName, _id, fields, doc) => {
     // check for fields collision
     else if (fields && item.fields) {
       const common = _.intersection(item.fields, fields)
-      if (common.length && DeepDiff(_.pick(item.doc, common), _.pick(doc, common))) {
-        console.log('Redis-Oplog: Handling potential race condition on ' + uid + ': ' + common.join(', '))
```
Comment (ramezrafla) [inline lib/redis/raceDetectionManager.js:47]: We do a diff after checking common.length as DeepDiff is expensive computationally

**Comment by ramezrafla** [comment]
Thanks for this. You made a lot of custom changes beyond the `transform` bit. Could you do a PR with minimal changes pls? If not, I'll simply copy over the couple of changes you made, but then you have to test again.

**Comment by fumbleforce** [comment]
Hi! Yeah I can do that, thanks for following up with your comments.

File: lib/mongo/mutator.js [hunk 2]
```diff
@@ -84,13 +84,15 @@ export default class Mutator {
...
     try {
       // we do this because when we send to redis
       // we need the exact _ids
       // and we extend the selector, because if between finding the docIds and updating
       // another matching insert sneaked in, its update will not be pushed
-      const updateSelector = _.extend({}, selector, {_id: { $in: docIds } })
+      const updateSelector = {
+        ...(typeof selector === "string" ? { _id: selector } : selector),
+        _id: { $in: docIds }
+      };
```
Comment (fumbleforce) [inline lib/mongo/mutator.js:95]: Do you mind this change?  Currently it fails if the selector is a string

**Pull request closed without merging**